### PR TITLE
TST: add a regression test for gh-16548

### PR DESCRIPTION
**Pull request by neutrinoceros**
### Description
Adding a regression test to make sure this already solved issue doesn't come back in the future.

- [ ] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

### CODE REVIEW

**Review by eerovaher — COMMENTED** [review]
I cherry-picked the commit here on top of the `v6.0.1` tag (78cf5340ddf9fe9731906224a638c4888849c93f) and the new test does fail with `ValueError: output array is read-only`. So this is indeed a valid regression test for a bug that has been present in older versions of `astropy`.

The code in the test comes from https://github.com/astropy/astropy/issues/16548#issuecomment-2155806283 by @ayshih, so maybe he should be the author of the commit and @neutrinoceros would only be the committer.

**Review by mhvk — APPROVED** [review]
Thanks, that looks good.

File: astropy/coordinates/tests/test_skyoffset_transformations.py [hunk 1]
```diff
@@ -366,3 +366,11 @@ def test_skyoffset_two_frames_interfering():
...
     # The line below was almost guaranteed to fail.
     dirs_icrs.transform_to(target_icrs.skyoffset_frame())
+
+
+def test_skyoffset_wrapping_vs_broadcasting():
+    # regression test for https://github.com/astropy/astropy/issues/16548
+    time = Time(["2021-01-01", "2021-01-01"])
+    origin = HCRS(0 * u.deg, 0 * u.deg, obstime=time)
+    frame = SkyOffsetFrame(origin=origin)
+    SkyCoord(240 * u.deg, 0 * u.deg, frame=frame)
```
Comment (pllim) [inline astropy/coordinates/tests/test_skyoffset_transformations.py:376]: Belated comment: Should we check the output of `SkyCoord` here?

Comment (neutrinoceros) [reply astropy/coordinates/tests/test_skyoffset_transformations.py:376]: I don't think it's necessary here but I'll defer to @ayshih

**Pull request merged**